### PR TITLE
PIM-5651: add migration for versioning

### PR DESCRIPTION
**Pull request by juliensnz**
**Description (for Contributor and Core Developer)**

In 1.5 we moved model class to the catalog component. As the versioning rely on the FQCN to create and find versions we need to update the version table with the correct FQCN

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | NA
| Added Behats                      | NA
| Changelog updated                 | N
| Review and 2 GTM                  | N
| Micro Demo to the PO (Story only) | N
| Migration script                  | Y
| Tech Doc                          | NA

### CODE REVIEW

**Comment by jjanvier** [comment]
gtm

**Comment by SamirBoulil** [comment]
gtm

File: upgrades/schema/Version_1_5_20160405165700_versioning.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Pim\Upgrade\schema;
+
+use Doctrine\DBAL\Migrations\AbstractMigration;
+use Doctrine\DBAL\Schema\Schema;
+
+/**
+ * In 1.5 we moved model class to the catalog component. As the versioning rely on the FQCN
+ * to create and find versions we need to update the version table with the correct FQCN
```
Comment (phaseinducer) [inline upgrades/schema/Version_1_5_20160405165700_versioning.php]: :+1:

File: upgrades/schema/Version_1_5_20160405165700_versioning.php [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    public function up(Schema $schema)
+    {
+        $movedEntities = [
+            'Pim\\\\Bundle\\\\CatalogBundle\\\\Model\\\\Product'       => 'Pim\\\\Component\\\\Catalog\\\\Model\\\\Product',
+            'Akeneo\\\\Bundle\\\\BatchBundle\\\\Entity\\\\JobInstance' => 'Akeneo\\\\Component\\\\Batch\\\\Model\\\\JobInstance',
+        ];
+
+        foreach ($movedEntities as $source => $target) {
+            $this->addSql(
+                'UPDATE pim_versioning_version SET resource_name = "' . $target . '" WHERE resource_name = "' . $source . '"'
```
Comment (phaseinducer) [inline upgrades/schema/Version_1_5_20160405165700_versioning.php]: sprintf would have been cleaner =)

**Comment by phaseinducer** [comment]
GTM

**Comment by willy-ahva** [comment]
migration = standard in 1.4

File: upgrades/schema/Version_1_5_20160405165700_versioning.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                'Pim\\\\Bundle\\\\CatalogBundle\\\\Model\\\\Product'       => 'Pim\\\\Component\\\\Catalog\\\\Model\\\\Product',
+                'Akeneo\\\\Bundle\\\\BatchBundle\\\\Entity\\\\JobInstance' => 'Akeneo\\\\Component\\\\Batch\\\\Model\\\\JobInstance',
+            ];
+
+            foreach ($movedEntities as $source => $target) {
+                $this->addSql(
+                    'UPDATE pim_versioning_version SET resource_name = "' . $target . '" WHERE resource_name = "' . $source . '"'
+                );
+            }
+        }
```
Comment (jjanvier) [inline upgrades/schema/Version_1_5_20160405165700_versioning.php]: else
or use the method `postup` like here https://github.com/akeneo/pim-enterprise-standard/blob/1.4/upgrades/schema/1.2-1.3/Version_1_3_20141203150000_product_statuses.php#L40

**Comment by jjanvier** [comment]
not gtm atm

**Comment by jjanvier** [comment]
gtm

**Pull request merged**